packages/js/assets/user-feedback-form.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * todo: this js file (bundled with the html file) should either:
+ * - come from the server (through a GET request)
+ * - or bundled in the js package as a post-build step (? - not yet explored)
+ */
+(function (window, document) {
+  // eslint-disable-next-line quotes
+  const TEMPLATE = `$$TEMPLATE$$`
+  const ENDPOINT = 'https://api.honeybadger.io/v1/feedback'
+
+  const HoneybadgerUserFeedbackForm = function () {};
+  HoneybadgerUserFeedbackForm.prototype.build = function () {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this
+    self.element = document.createElement('div')
+    self.element.id = 'honeybadger-feedback-wrapper'
+    self.element.innerHTML = TEMPLATE
+
+    document.body.appendChild(self.element)
+
+    self.form = self.element.getElementsByTagName('form')[0]
+    self.form.addEventListener('submit', function (e) {
+      e.preventDefault()
+      self.submit()
+    })
+
+    const cancelButton = document.getElementById('honeybadger-feedback-cancel')
+    cancelButton.onclick = (e) => {
+      e.preventDefault()
+      self.close()
+    }
+  };
+
+  HoneybadgerUserFeedbackForm.prototype.close = function () {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this
+    self.element.parentNode.removeChild(self.element);
+  };
+
+  HoneybadgerUserFeedbackForm.prototype.submit = function () {
+    // eslint-disable-next-line @typescript-eslint/no-this-alias
+    const self = this
+    if (self.loading) return
+    self.loading = true
+
+    window.fetch(ENDPOINT +
+        '?format=js' +
+        `&token=${self.getLastNoticeId()}` +
+        `&name=${encodeURIComponent(self.form.name.value)}` +
+        `&email=${encodeURIComponent(self.form.email.value)}` +
+        `&comment=${encodeURIComponent(self.form.comment.value)}`,
+    {
+      method: 'GET',
+      // todo: this should not be here
+      // server should respond with header "Access-Control-Allow-Origin: CALLER_HOST"
```

Comment:
We currently only serve CORS headers on the API instances from the `/v1/notices/js` path.

Reply:
> > I tried not to apply custom styling so that it would be adapted based on the user's CSS.
> 
> I appreciate not wanting to push our own styling, but the mixture of serif (most text) and sans-serif (buttons) doesn't look great, and Times New Roman is meh. :) How about something like this (from [CSS tricks](https://css-tricks.com/snippets/css/system-font-stack/)):
> 
> ```css
> #honeybadger-feedback {
>   font-family: system-ui, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
> }
> ```

Awesome, thanks!

Reply:
@subzero10 I don't recall, but this feature was implemented a long time ago, so it could be a browser compatibility thing. :grin:

Reply:
OK thanks! I guess we don't have to introduce any changes since it works.
How about authentication though? Shouldn't we go for it?

Reply:
> OK thanks! I guess we don't have to introduce any changes since it works.
> How about authentication though? Shouldn't we go for it?

I thought it *was* authenticated with the API key, but yes, authentication sounds reasonable. :grin:

Reply:
Nope, it's not. All you need is the notice id!

Reply:
https://github.com/honeybadger-io/honeybadger/issues/3597